### PR TITLE
Add bundle size limits and tracking to Github CLI

### PR DESCRIPTION
**Pull request by arafatkatze**
Add size limit enforcement and historical tracking for extension bundles to prevent unexpected size increases. This helps maintain performance and download speeds for users.
Part of the linear [issue for Cody Performance](https://linear.app/sourcegraph/issue/CODY-4148/orthogonal-performance-and-quality-metrics-for-cody-extension)

The purpose of this PR is to have bundle sizing measurements for both the extension and the webview. After having them, integrate them into the CI. If you were to add a dependency that increases the bundle size by a lot, it will fail the CI, therefore you have to be cautious about making reckless changes. If you really have to do it, making changes in the CI would force you to make potential changes with extension bloating carefully. NOTE the limit is set to 10% extra so its unlikely you will hit that limit unless you do some really reckless changes. 

NOTE: This PR also puts the size of the bundle in bigquery on every release 
image

This PR also tracks the bundle sizes on every pre-release along with the date so we have a record of the time when small changes happened and we can then use the bigquery table to see how the bundle sizes evolved. 

## Test plan
- Actually run the Github Actions CI 
The failure mode looks like this and you see this failure mode on the build step so when you do a build and it makes very large bundle then the build fails and you have to talk to team cody core to get permission from Grace/dom before making big changes but for small bundle changes this will pass. 

image

- How did I check if this works with the release github actions? 
Since I couldn't actually run the release github action as that would make the release I decided to make a close approximation in another github action  which would entail getting the release version and then making a bigquery request to update a row in the table to mark the date and the sizes of the bundles 

image
This is the response of my fake requests 

This is the [workflow file](https://github.com/sourcegraph/cody/actions/runs/12233046828/workflow?pr=6222) and this is [the run](https://github.com/sourcegraph/cody/actions/runs/12233046828/job/34119311775?pr=6222).

### CODE REVIEW

File: .github/workflows/vscode-insiders-release.yml [hunk 1]
```diff
@@ -22,6 +22,45 @@ jobs:
...
+        with:
+          version: 'latest'
+          service_account_key: ${{ secrets.GCP_BQ_CREDENTIALS }}
+          project_id: 'cody-core-dev'
+          export_default_credentials: true
+
+      - name: Insert data into BigQuery
+        run: |
+          # Write the service account key to a file
+          echo '${{ secrets.GCP_BQ_CREDENTIALS }}' > gcloud-service-key.json
```
Comment (kalanchan) [inline .github/workflows/vscode-insiders-release.yml:46]: @sourcegraph/security-code-review can I get a review here?

Ideally I don't think we should be storing the service account keys locally, what is the risk here? 

can we use something like https://github.com/google-github-actions/auth instead

Comment (kalanchan) [reply .github/workflows/vscode-insiders-release.yml:46]: @andreeleuterio

Comment (andreeleuterio) [reply .github/workflows/vscode-insiders-release.yml:46]: yes, thanks for the great callour @kalanchan. You can find example [here](https://github.com/google-github-actions/auth/blob/main/docs/EXAMPLES.md#examples-for-authenticating-to-google-cloud-from-github-actions) and GCP workload identity [here](https://github.com/google-github-actions/auth?tab=readme-ov-file#setup). @arafatkatze lmk if you have trouble here.

Comment (arafatkatze) [reply .github/workflows/vscode-insiders-release.yml]: Thanks for the super important callout. I fixed this and tested it too.

Comment (arafatkatze) [reply .github/workflows/vscode-insiders-release.yml]: @kalanchan @andreeleuterio Can you take a look again?

Comment (andreeleuterio) [reply .github/workflows/vscode-insiders-release.yml]: @arafatkatze not quite. You are using the correct gcloud auth action, but with the [Service Account Key JSON](https://github.com/google-github-actions/auth?tab=readme-ov-file#service-account-key-json) setup. We want the  [Direct Workload Identity Federation option](https://github.com/google-github-actions/auth?tab=readme-ov-file#preferred-direct-workload-identity-federation). From what I see in the instructions, just get Entitle permissions for the GCP project in question and you should be able to follow the steps. If you run into issues let me know.

File: .github/workflows/bundle-size-tracker.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+name: Bundle Size Tracker
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+permissions:
+  id-token: write
```
Comment (andreeleuterio) [inline .github/workflows/bundle-size-tracker.yaml:9]: @arafatkatze can we remove them from here now?

File: .github/workflows/vscode-insiders-release.yml [hunk 3]
```diff
@@ -22,6 +22,37 @@ jobs:
...
+          pnpm ts-node ./scripts/measure-bundle-size.ts
+          cd ..
+          echo "Extension Bundle Size: $EXTENSION_BUNDLE_SIZE_MB MB"
+          echo "Webview Bundle Size: $WEBVIEW_BUNDLE_SIZE_MB MB"
+        env:
+          GITHUB_ENV: $GITHUB_ENV
+      - name: Authenticate to Google Cloud
+        uses: google-github-actions/auth@v1
+        with:
+          credentials_json: ${{ secrets.GCP_BQ_CREDENTIALS }}
```
Comment (andreeleuterio) [inline .github/workflows/vscode-insiders-release.yml]: remember to update it here as well 😅

**Review by andreeleuterio — CHANGES_REQUESTED** [review]

File: .github/workflows/vscode-insiders-release.yml (hunk 3 above)
Comment (arafatkatze) [reply .github/workflows/vscode-insiders-release.yml]: Done

File: .github/workflows/bundle-size-tracker.yaml (hunk 2 above)
Comment (arafatkatze) [reply .github/workflows/bundle-size-tracker.yaml:9]: No we need both the workflow level and the job level

File: .github/workflows/vscode-insiders-release.yml (hunk 1 above)
Comment (arafatkatze) [reply .github/workflows/vscode-insiders-release.yml]: Thanks to help from @andreeleuterio we are using [Direct Workload Identity Federation option](https://github.com/google-github-actions/auth?tab=readme-ov-file#preferred-direct-workload-identity-federation).

File: .github/workflows/bundle-size-tracker.yaml (hunk 2 above)
Comment (arafatkatze) [reply .github/workflows/bundle-size-tracker.yaml]: No you are right, I got rid of it.

**Review by andreeleuterio — APPROVED** [review]

**Pull request merged**